addons/mrp/stock.py

```diff
@@ -61,7 +61,7 @@ def _action_explode(self, cr, uid, move, context=None):
                     mid = move_obj.copy(cr, uid, move.id, default=valdef, context=context)
                     to_explode_again_ids.append(mid)
                 else:
-                    if prod_obj.type in ('consu','product'):
+                    if product.type in ('consu','product'):
```

Comment:
Please check 53 line, because it's the same condition here too, so this condition never will be done.